arbor/include/arbor/cable_cell.hpp

```diff
@@ -349,7 +285,29 @@ class cable_cell {
     const cable_cell_region_map& region_assignments() const;
     const cable_cell_location_map& location_assignments() const;
 
+    // View the decorations on the cell.
+    const decor& decorations() const {
+        return decor_;
+    }
+
+    const cable_cell_parameter_set& default_parameters() const {
+        return decor_.defaults;
+    }
+
```

Comment:
Is it necessary to have the `decorations()` method? Following on from the comment above on `cable_cell::decorate`, `default_parameters()`, like `region_assignments()` and `location_assignments()`, could be delegated to the pimpl object, and then there would be no need to hold on to a copy of the `decor_` object at all.

Reply:
One reason to keep it around (and to keep morphology and label dictionary information around) is to make it possible to "serialize" and `cable_cell`, which is straightforward with the decor, morphology and label_dict used to instantiate it.

Whether we want to do this is another matter. You could argue that the original decor, labels and morphology used to create the cell can be retained and used for that purpose. I haven't put enough thought into possible use cases and situations that could arise.

Reply:
I was wondering about the serialization interface. For descriptions, I agree that it's the decor that should be serialized, rather than the cell. For snapshot purposes, it's open: do we represent cells, or do we represent cell groups? If the former, then I think it should be in terms of thingified quantities; if the latter, then the vectors of stuff based on the discretization.

Reply:
I was referring to descriptions, and not cell state.

I agree that the decor + morphology + labels should be the inputs for serialization (that's the whole point of this refactoring, anyway). The question is whether one should be able to "extract" those three things from the `cable_cell` for serialization, or it whether the inputs used to create the cell should be retained for that purpose.

One use case I can think of would be serializing a "complete" cell description generated from a recipe, where the inputs used to generate the cell are not available (e.g. the recipe comes from a library or a python call back).

```C++
arb::cable_cell c = std::any_cast<arb::cable_cell>(rec.get_cell_description(42));
serialize(c);
```

This seems like a reasonable thing to support, however we can't do it at the moment because `cable_cells` don't retain the `label_dict`, and if the morphology is retained, it is kept in the bottom of a provider somewhere that isn't obvious.